### ISSUE TITLE
[Bug] Auto-Capitalization of Header Keys

### ISSUE THREAD

**Issue by epraetor-eolian** [yesoreyeram/grafana-infinity-datasource#444]
## **Describe the bug**

When you enter in headers in either the data source or the query panel, the header key gets capitalized and this makes the query not work if the API you are querying is on the finicky side. For instance, if there is a required header called 'required-header' it gets changed in the actual query to 'Required-Header' and this makes queries not work for an API that I am using.

When I copy the cURL text and run it in command, the query at first does not work properly, but once I remove the capitalization from this header it works fine. Not sure if this is an issue with Grafana or the plugin itself.

## **Steps to reproduce**

Not able to share credentials for the exact API that I am using, but if you enter in either an auth header or other custom header in the datasource or panel query, and then use the query inspector to view the cURL text, you will see that those fields get capitalized.

**Note:** bugs will be fixed only if enough information provided to reproduce the issue

## **Sample Data**

Provide a [link](https://gist.github.com) for sample data, or provide sample data bellow

## **Screenshots**

![image](https://user-images.githubusercontent.com/101751469/198129301-f06022a4-eb09-4604-a658-17221173c020.png)

## **Version Details:**

- Grafana version : 9.2.2
- Plugin version : 1.0.0

**Comment by epraetor-eolian** [comment]
ok thanks for the response, I'll keep looking into it

**Comment by epraetor-eolian** [comment]
Agreed that API should be case insensitive for header names. I actually reached out to the API owners to get this implemented. Thanks for taking a second look at this - something to keep in mind that can happen but not sure if worth the effort to implement a workaround

**Comment by epraetor-eolian** [comment]
One other note with the generated CURL text is that there appears to be an extra, unmatched quotation mark at the end that isn't matched. When I run in cmd it throws an error, and then when I delete this extra quote mark the command works as it should. 

Not sure if you see the same thing in your generated CURL

**Comment by shahid430** [comment]
We configured infinity plugin for our application but there is no option for access key & secret key to be applied. We are using "X-Auth_token" to authenticate but the token expiry is 24 hours only, after which we need to reset the header with new token.

Could someone please provide the configuration file for secret & access key to get rid of manually entering the X-Auth-Token everyday.
